aws/resource_aws_cognito_resource_server_test.go

```diff
@@ -0,0 +1,149 @@
+package aws
+
+import (
+	"errors"
+	"fmt"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/cognitoidentityprovider"
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestAccAWSCognitoResourceServer_basic(t *testing.T) {
+	identifier := fmt.Sprintf("tf-acc-test-resource-server-id-%s", acctest.RandStringFromCharSet(10, acctest.CharSetAlphaNum))
+	name := fmt.Sprintf("tf-acc-test-resource-server-name-%s", acctest.RandStringFromCharSet(10, acctest.CharSetAlphaNum))
+	poolName := fmt.Sprintf("tf-acc-test-pool-%s", acctest.RandStringFromCharSet(10, acctest.CharSetAlphaNum))
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSCognitoResourceServerDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSCognitoResourceServerConfig_basic(identifier, name, poolName),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccCheckAWSCognitoResourceServerExists("aws_cognito_resource_server.main"),
+					resource.TestCheckResourceAttr("aws_cognito_resource_server.main", "identifier", identifier),
+					resource.TestCheckResourceAttr("aws_cognito_resource_server.main", "name", name),
+					resource.TestCheckResourceAttr("aws_cognito_user_pool.main", "name", poolName),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAWSCognitoResourceServer_full(t *testing.T) {
```

Comment:
We should descriptively name this `_scopes` to signify what its actually testing